src/jit/codegenarm64.cpp

```diff
@@ -4578,49 +4578,139 @@ void CodeGen::genSIMDIntrinsicGetItem(GenTreeSIMD* simdNode)
     // - the source of SIMD type (op1)
     // - the index of the value to be returned.
     genConsumeOperands(simdNode);
-    regNumber srcReg = op1->gtRegNum;
 
-    // TODO-ARM64-CQ Optimize SIMDIntrinsicGetItem
-    // Optimize the case of op1 is in memory and trying to access ith element.
-    assert(op1->isUsedFromReg());
+    emitAttr baseTypeSize  = emitTypeSize(baseType);
+    unsigned baseTypeScale = genLog2(EA_SIZE_IN_BYTES(baseTypeSize));
 
-    emitAttr baseTypeSize = emitTypeSize(baseType);
-
-    if (op2->IsCnsIntOrI())
+    if (op1->isContained())
     {
-        assert(op2->isContained());
+        // Optimize the case of op1 is in memory and trying to access ith element.
+        assert(!op1->isUsedFromReg());
+
+        if (op1->OperIsLocal())
+        {
+            unsigned varNum = op1->gtLclVarCommon.gtLclNum;
+
+            if (op2->IsCnsIntOrI())
+            {
+                assert(op2->isContained());
 
-        ssize_t index = op2->gtIntCon.gtIconVal;
+                ssize_t index = op2->gtIntCon.gtIconVal;
 
-        if (getEmitter()->isValidVectorIndex(emitTypeSize(simdType), baseTypeSize, index))
+                if (getEmitter()->isValidVectorIndex(emitTypeSize(simdType), baseTypeSize, index))
+                {
+                    // Only generate code for the get if the index is valid
+                    // Otherwise generated code will throw
```

Comment:
Here

Reply:
Okay - I read this as "if we generate code for the get, the generated code will throw, so we shouldn't do that", so perhaps it would be clearer as "We only need to generate code for the get if the index is valid, otherwise the (previously generated) code for the check will throw."